front/lib/tracking/customerio/server.ts

```diff
@@ -0,0 +1,175 @@
+import type {
+  LightWorkspaceType,
+  MembershipRoleType,
+  UserType,
+  UserTypeWithWorkspaces,
+} from "@dust-tt/types";
+import * as _ from "lodash";
+
+import { subscriptionForWorkspace } from "@app/lib/auth";
+import { Workspace } from "@app/lib/models/workspace";
+import { MembershipResource } from "@app/lib/resources/membership_resource";
+import { renderLightWorkspaceType } from "@app/lib/workspace";
+
+const {
+  CUSTOMERIO_ENABLED,
+  CUSTOMERIO_SITE_ID = "",
+  CUSTOMERIO_API_KEY = "",
+} = process.env;
+
+const CUSTOMERIO_HOST = "https://track-eu.customer.io/api/v2";
+
+function isCustomerioEnabled() {
+  return CUSTOMERIO_ENABLED === "true";
+}
+
+export class CustomerioServerSideTracking {
+  static trackSignup({ user }: { user: UserType }) {
+    if (!isCustomerioEnabled()) {
+      return;
+    }
+    return CustomerioServerSideTracking._identifyUser({ user });
+  }
+
+  static async trackUserMemberships({
+    user,
+  }: {
+    user: UserTypeWithWorkspaces;
+  }) {
+    if (!isCustomerioEnabled()) {
```

Comment:
I think we should do the `isCustomerioEnabled()` check only at the transport layer. We still want to know if something is wrong with this code in dev, we just don't want to fire the HTTP query.
Can we remove this check and only keep the one in transport functions?

Reply:
Done